share/picongpu/tests/Pusher/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,387 @@
+    x/y_poss (np.array): Relative in-Cell positions of the particle in the
+                         series
+    x/y_offSet (np.array): All knot-positions of the particle in the series
+    periode (int): openPMD periode of the simulation
+    charge (float): Charge of particle
+    mass (float): Mass of particle
+    x/y_momentum (float): All momenta of the particle in the series
+    R_c (float): Radius of gyration of the particle at timestep 0
+
+    """
+    num_iter = len(series.iterations)
+
+    # create arrays for saving the data
+
+    # in-cell position
+    x_poss = np.zeros(num_iter)
+    y_poss = np.zeros(num_iter)
+
+    # cells
+    x_offSet = np.zeros(num_iter)
+    y_offSet = np.zeros(num_iter)
+
+    # momentum
+    x_momentum = np.zeros(num_iter)
+    y_momentum = np.zeros(num_iter)
+
+    i = series.iterations[0]
+    # charge
+    charge = i.particles["e"]["charge"][io.Mesh_Record_Component.SCALAR]
+    # mass
+    mass = i.particles["e"]["mass"][io.Mesh_Record_Component.SCALAR]
+
+    # calculating of the openPMD-periode for later use
+    indices = np.array(series.iterations)
+    periode = indices[1] - indices[0]
+
+    for iteration in indices:
+        i = series.iterations[iteration]  # current position
+
+        # node coordinates
+        pos2_x = i.particles["e"]["position"]["x"].load_chunk()
+        pos2_y = i.particles["e"]["position"]["y"].load_chunk()
+
+        # InCell coordinates
+        pos_off2_x = i.particles["e"]["positionOffset"]["x"].load_chunk()
+        pos_off2_y = i.particles["e"]["positionOffset"]["y"].load_chunk()
+
+        p_x = i.particles["e"]["momentum"]["x"].load_chunk()
+        p_y = i.particles["e"]["momentum"]["y"].load_chunk()
+
+        series.flush()
+
+        array_index = int(iteration/periode)
+        # save the read data in arrays for later usage
+        x_poss[array_index] = pos2_x
+        y_poss[array_index] = pos2_y
+
+        x_offSet[array_index] = pos_off2_x
+        y_offSet[array_index] = pos_off2_y
+
+        x_momentum[array_index] = p_x
+        y_momentum[array_index] = p_y
+
+    return (x_poss, y_poss, x_offSet, y_offSet, x_momentum, y_momentum,
+            charge, mass, num_iter, periode)
+
+
+def compare_momentum(x_momentum, y_momentum,
+                     charge, mass, B, params, num_iter, periode):
+    """Tests if the change in radius from one revolution to the other is
+    greater than epsilon = 1e-5.
+    The radii are calculated by the momenta (and q*B) in the x-y-plane.
+
+    Parameters:
+    x/y_momentum(array): Series of the momenta of the particle in x/y-direction
+    charge (float): Charge of the particle
+    mass (float): Mass of the particle
+    B (float): Magnitude of the magnetic field B in the test
+    params (list): Simulation parameters
+    num_iter (int): Number of iterations in the series
+    period (int): openPMD period of the simulation
+
+    Returns:
+    0: if the radius of gyration does not vary much (<epsilon)
```

Comment:
As I understand it: The CI works with 0 as true and every other integer as false. Therefore I use these return values. Also, we can distinguish the different Errors by their non-zero number.

Reply:
in this case use an [enumeration](https://docs.python.org/3/library/enum.html), this will allow you to write meaningful names in the code instead of just integers and for the CI yo can covnert them to an integer if required.